src/index.js

```diff
@@ -218,7 +218,9 @@ let currentComponent;
 
 
 function createFactory(type) {
-	return createElement.bind(null, type);
+	return function () {
+		return createElement.apply(type, arguments);
```

Comment:
Not sure why the tests aren't failing, but I think this is missing an argument?

```js
createElement.apply('div', 'a', { href: '#foo' })
```
That would seem to be setting `this` to `"div"` in createElement.

Reply:
Yes, I guess this implementation is wrong.
I could think of some ugly but correct implementation such as ```createElement.apply(null, [type].concat(Array.apply(null, arguments)))```.

In general, that's probably a bad idea and should not be merged. I will send another pr if I come up with some better idea.